src/server_manager/web_app/ui_components/outline-per-key-data-limit-dialog.ts

```diff
@@ -0,0 +1,380 @@
+ * limit of the UI key, and the actual data limit as saved on the server.  App controls the UI data
+ * limit and the request to the server, and the display key in the element is never itself changed.
+ */
+@customElement('outline-per-key-data-limit-dialog')
+export class OutlinePerKeyDataLimitDialog extends LitElement {
+  static get styles() {
+    return [
+      COMMON_STYLES,
+      css`
+        #container {
+          width: 100%;
+          display: flex;
+          flex-flow: column nowrap;
+        }
+
+        #dataLimitIcon {
+          /* Split the padding evenly between the icon and the section to be bidirectional. */
+          padding: 0 12px;
+        }
+
+        #headerSection {
+          display: flex;
+          flex-direction: row;
+          padding: 0 12px;
+        }
+
+        #headerSection h3 {
+          font-size: 18px;
+          font-weight: 500;
+          color: rgba(0, 0, 0, 0.87);
+          line-height: 24px;
+        }
+
+        #menuSection {
+          flex: 1;
+          padding: 0 78px;
+          margin-top: 10px;
+        }
+
+        #buttonsSection {
+          margin-top: 10px;
+          display: flex;
+          flex-direction: row-reverse;
+        }
+
+        paper-button {
+          display: flex;
+          height: 36px;
+          text-align: center;
+        }
+
+        #save {
+          background-color: var(--primary-green);
+          color: #fff;
+        }
+
+        #save[disabled] {
+          color: var(--dark-gray);
+          background-color: rgba(0, 0, 0, 0.13);
+        }
+
+        #menu {
+          display: flex;
+          flex-flow: row nowrap;
+        }
+
+        #unitsDropdown {
+          width: 50px;
+          padding: 0 10px;
+        }
+
+        paper-checkbox {
+          /* We want the ink to be the color we're going to, not coming from */
+          --paper-checkbox-checked-color: var(--primary-green);
+          --paper-checkbox-checked-ink-color: var(--dark-gray);
+          --paper-checkbox-unchecked-color: var(--dark-gray);
+          --paper-checkbox-unchecked-ink-color: var(--primary-green);
+        }
+
+        paper-listbox paper-item:hover {
+          cursor: pointer;
+          background-color: var(--background-contrast-color);
+          color: #fff;
+        }
+      `,
+    ];
+  }
+
+  /**
+   * @member _keyName The displayed name of the UI access key representing the key we're working on.
+   */
+  @internalProperty() _keyName = '';
+  /** @member _keyId The id of the UI access key representing the key we're working on. */
+  @internalProperty() _keyId = '';
+  /**
+   * @member _keyDataLimitBytes The data limit, if it exists, on the access key we're working on.
+   */
+  @internalProperty() _keyDataLimitBytes: number = undefined;
+  /**
+   * @member serverDefaultLimitBytes The default data limit of the server we're working on, or null
+   */
+  @internalProperty() _serverDefaultLimitBytes: number = undefined;
```

Comment:
Remove this property.

Reply:
See below, I think it's possible to see either the bytes amount or the `DisplayDataAmount` as the derived property and it's cleaner to keep the formatting logic contained in the UI instead of split between the App and the dialog.